hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java

```diff
@@ -134,6 +136,56 @@ public void readWithTypeCoertion() {
                 )
         );
     }
+    @Test
+    public void readWithOneProcessor() {
+        MongoCollection<Document> collection = database.getCollection(collectionName);
+        collection.insertOne(new Document("firstName", "Luke").append("lastName", "Skywalker").append("jedi", "true"));
+
+        Pattern parallelismDagPattern = Pattern.compile("\\[localParallelism=(\\d)+]");
+        AtomicBoolean otherFound = new AtomicBoolean(false);
+        LogListener lookForProjectAndFilterStep = log -> {
+            String message = log.getLogRecord().getMessage();
+            Matcher matcher = parallelismDagPattern.matcher(message);
+            if (matcher.find()) {
+                String number = matcher.group(1);
+                if (!"1".equals(number)) {
+                    otherFound.set(true);
+                }
+            }
+        };
+        for (HazelcastInstance instance : instances()) {
+            instance.getLoggingService().addLogListener(Level.FINE, lookForProjectAndFilterStep);
+        }
+
+        execute("CREATE MAPPING " + collectionName
+                + " ("
+                + " id OBJECT external name _id, "
+                + " firstName VARCHAR, "
+                + " lastName VARCHAR, "
+                + " jedi BOOLEAN "
+                + ") "
+                + "TYPE MongoDB "
+                + "OPTIONS ("
+                + "    'connectionString' = '" + mongoContainer.getConnectionString() + "', "
+                + "    'database' = '" +  databaseName + "', "
+                + "    'forceMongoReadParallelismOne' = 'true' "
```

Comment:
Do we have test checking that the default behaviour is parallel

Reply:
Now we do ;)